### PR TITLE
Fixed CMake config files #41

### PR DESCRIPTION
**Pull request by willyd**
Other changes:
- Switch from nmake to ninja. Ninja is much faster than nmake or VS
- Explicitly disabled opencv_hdf and opencv_matlab

### CODE REVIEW

**Comment by jakirkham** [comment]
> Explicitly disabled opencv_hdf and opencv_matlab

Why would we want to do this?

**Comment by willyd** [comment]
> Why would we want to do this?

I assumed that since appveyor does not have MATLAB disabling it made sense.

The opencv_hdf component was failing to build and HDF5 is missing from the dependencies on windows so I thought that excluding this component was the original intent. Maybe it's best to include the dependency and fix the build of the component if the exclusion was not intentional.

That being said, neither are present in the current package so it does not change anything to the end result.

**Comment by jakirkham** [comment]
> HDF5 is missing from the dependencies on windows so I thought that excluding this component was the original intent.

This is probably a holdover from before we had HDF5 working (I'm guessing). @patricksnape, do you know anything about this?

**Comment by willyd** [comment]
Although it is in the title of the issue and commit message this fixes #41.

File: recipe/bld.bat [hunk 1]
```diff
@@ -109,6 +111,31 @@ if "%ARCH%" == "64" ( set "OPENCV_ARCH=64")
 robocopy %LIBRARY_PREFIX%\x%OPENCV_ARCH%\vc%VS_MAJOR%\ %LIBRARY_PREFIX%\ *.* /E
 if %ERRORLEVEL% GEQ 8 exit 1
 
+rem patch the cmake files accordingly
+"%PYTHON%" "%RECIPE_DIR%\patch_cmake_targets.py" ^
```
Comment (patricksnape) [inline recipe/bld.bat]: I would prefer if these were actually patches

Comment (willyd) [inline recipe/bld.bat]: Makes sense. Would something like in [vcpkg](https://github.com/Microsoft/vcpkg/blob/master/ports/opencv/opencv-installation-options.patch) be acceptable?

Comment (patricksnape) [inline recipe/bld.bat]: Of course, if that works correctly 👍

**Comment by patricksnape** [comment]
Possible reasons for failure:

http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1002

May need to either pin Eigen to last working version or try passing some flags on Visual Studio 2008 to disable SSE.

**Comment by jakirkham** [comment]
Are you still interested in this PR, @willyd?

**Comment by willyd** [comment]
I don't have much time to spend on this right now. If you wish to close the PR I don't mind. I agree with @patricksnape that using patches would be cleaner. I will resubmit another PR when I have more time if you choose to close the issue.

**Comment by giumas** [comment]
It is very annoying to have to manually patch the CMake files each time.. Any plan to just use the PR replacing approach while waiting for a better solution?

**Comment by jakirkham** [comment]
Not likely. We prefer patches in general. They are easier to track, easier to follow, and easier to submit upstream.

**Comment by giumas** [comment]
@jakirkham, the issue is related to the way that a Conda package organizes the folders, so it is unlikely that the patch will be accepted upstream.

**Pull request closed without merging**